Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py

```diff
@@ -551,6 +558,621 @@ def decode_url(headers: str) -> dict:
+                            event.get('attackData',  # type: ignore[attr-defined]
+                                      {}).get(attack_data_key, ""))
+                if "httpMessage" in event:
+                    event['httpMessage']['requestHeaders'] = decode_url(  # type: ignore[index]
+                        event.get('httpMessage', {}).get('requestHeaders', ""))  # type: ignore[attr-defined]
+                    event['httpMessage']['responseHeaders'] = decode_url(  # type: ignore[index]
+                        event.get('httpMessage', {}).get('responseHeaders', ""))  # type: ignore[attr-defined]
+            except Exception as e:
+                demisto.debug(f"Couldn't decode {event=}, reason: {e}")
+            finally:
+                processed_events.append(event)
+    demisto.info(f"Running in interval = {counter}. Sending {len(processed_events)} events to xsiam.")
+    tasks = send_events_to_xsiam_akamai(processed_events, VENDOR, PRODUCT, should_update_health_module=False,
+                                        chunk_size=SEND_EVENTS_TO_XSIAM_CHUNK_SIZE, send_events_asynchronously=True,
+                                        url_key="host", data_format="json", data_size_expected_to_split_evenly=True,
+                                        counter=counter)
+    demisto.info(f"Running in interval = {counter}. Finished executing send_events_to_xsiam, waiting for tasks to end.")
+    await asyncio.gather(*tasks)
+    demisto.info(f"Running in interval = {counter}. Finished gathering all tasks.")
+    demisto.info(f"Running in interval = {counter}. Updating module health.")
+    set_integration_context({"offset": offset})
+    demisto.updateModuleHealth({'eventsPulled': len(processed_events)})
+    demisto.info(f"Running in interval = {counter}. Finished updating module health.")
+
+
+async def get_events_from_akamai(client: Client,
+                                 config_ids: str,
+                                 from_time: str,
+                                 page_size: int,
+                                 offset: str | None,
+                                 max_concurrent_tasks: int):
+    """Iteratively checks for condition where the loop should wait due to multiple reasons
+    and then gathers events from Akamai SIEM.
+
+    Args:
+        client: Client object with request.
+        config_ids (str): security configuration ids to fetch, e.g. `51000;56080`.
+        from_time (str): From when to fetch if first time, e.g. `3 days`.
+        page_size: The number of events to limit for every request.
+        offset (str | None): The offset (hash) to use for offset based mechanism.
+        max_concurrent_tasks (int): The maximum number of tasks allowed to run concurrently.
+
+    Yields:
+        (list[str], int, str): events (not serialized), counter (the current execution number), and the new offset.
+    """
+    counter = 0
+    while True:
```

Comment:
will any exceptions break out of this generator loop? http errors or anything at all? lets make sure this loop always restarts

Reply:
we catch this errors from inside the code and announce them using demisto.error, module health with is_error=True and sleep.